tracer/src/Datadog.Trace/LibDatadog/LibDatadogAvailaibilityHelper.cs

```diff
@@ -0,0 +1,64 @@
+// <copyright file="LibDatadogAvailaibilityHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System;
+using Datadog.Trace.ClrProfiler;
+using Datadog.Trace.ClrProfiler.ServerlessInstrumentation;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.LibDatadog;
+
+/// <summary>
+/// *This class should NOT contain any direct Logger field, nor methods should log*.
+/// LibDatadogAvailable factory is used when building settings for the tracer, if a logger is instantiated in this path, it creates an infinite loop as the logger itself will try to build settings
+/// The Lazy will loop on itself and end up in a InvalidOperationException as Value ends up calling itself
+/// Even calling a method on a class that has a Logger as a static field is enough to have it instantiated, that's why ProfilerAttached eventually call FrameworkDescription that has a Lazy`Logger`
+/// </summary>
+internal static class LibDatadogAvailaibilityHelper
+{
+    // This will never change, so we use a lazy to cache the result.
+    // This confirms that we are in an automatic instrumentation environment (and so P/Invokes have been re-written)
+    // and that the libdatadog library has been deployed (which is not the case in some serverless environments).
+    // We should add or remove conditions from here as our deployment requirements change.
+    private static readonly Lazy<LibDatadogAvailableResult> LibDatadogAvailable = new(() =>
+    {
+        var res = IsServerlessEnvironment();
+        var isServerless = res.Item1;
+        var possibleException = res.Item2;
+        return new(!isServerless && Instrumentation.ProfilerAttached, possibleException);
```

Comment:
`Instrumentation.ProfilerAttached` only check that we loaded `Datadog.Trace.ClrProfiler.Native.dll` or OS equivalent. Should this check that we loaded   `datadog_profiling_ffi.dll` to be safe?

Just checking "is serverless" based on env vars seems dangerous if we get it wrong. What happens if we think `libdatadog` is available but the file is not actually there?

Reply:
(This is not specific to this PR, I've brought it up several times before.)

Reply:
yes maybe we could do the same as `ProfilerAttached` and expose `IsLibdatadogAvailable` as [the test already exists in native](https://github.com/DataDog/dd-trace-dotnet/blob/ce361d118c49a2782db5644bcde134212bf3e5cb/tracer/src/Datadog.Tracer.Native/cor_profiler.cpp?plain=1#L965C1-L965C64) and we could interop expose it  but maybe not in this PR 😅